week2/homework/src/index.js

```diff
@@ -1,3 +1,118 @@
 'use strict';
+let fs = require('fs');
+let order = process.argv[2];
+let orderDefinition = process.argv[3];
+let orderDefinition2 = process.argv[4];
 
-// TODO: Write the homework code in this file
+fs.readFile('./toDo.json', 'utf8', function (error, data) {
+  if (error) {
+    console.log(error);
+  }
+  else {
+    let myToDoList = JSON.parse(data);
+
+    function help() {
+      fs.readFile('./help.txt', 'utf8', function (error, data) {
+        if (error) {
+          console.log(error);
+        }
+        else {
+          console.log(data);
+        }
+      }
+      );
+    };
+
+    function list() {
+      function isEmpty(myToDoList) {
+        if (Object.values(myToDoList).length === 0) {
+          return true;
+        }
+        return false;
+      }
+      if (isEmpty(myToDoList)) {
+        console.log('You have not any plan in your To Do List');
+      }
+      else {
+        console.log(myToDoList);
+      }
+    };
+
+    function add(orderDefinition, orderDefinition2) {
+      let toDoDate = orderDefinition2.toLocaleDateString();
+      myToDoList[Object.keys(myToDoList).length + 1] = orderDefinition + ' ' + toDoDate;
+      fs.writeFile(
+        './toDo.json',
+        JSON.stringify(myToDoList),
+        (error) => {
+          if (error) { console.log(error); }
+        });
+    };
+
+    function remove(orderDefinition) {
+      if (orderDefinition > 0) {
+        delete myToDoList[orderDefinition];
```

Comment:
In the help section it says `Removes a to-do item by its 1-base index.`, but this is not what's happening, am I wrong? It's using the 0-based index, so it doesn't do what the user expects.
Correct me if I'm wrong :)

Reply:
I think that is not actually,  the code uses "const index = Object.keys(myToDoList).length + 1;" So it always starts from "1".  If we reset the list and add sth, we can see { '1': 'sth" }

Reply:
You are absolutely right! my fault 🙇